pkg/endpoint/events.go

```diff
@@ -119,6 +120,97 @@ func (e *Endpoint) PolicyRevisionBumpEvent(rev uint64) {
 	}
 }
 
+/// EndpointNoTrackEvent contains all fields necessary to update the NOTRACK rules.
+type EndpointNoTrackEvent struct {
+	ep     *Endpoint
+	annoCB AnnotationsResolverCB
+}
+
+// Handle handles the NOTRACK rule update.
+func (ev *EndpointNoTrackEvent) Handle(res chan interface{}) {
+	var oldPort uint16
+
+	e := ev.ep
+	port, err := ev.annoCB(e.K8sNamespace, e.K8sPodName)
```

Comment:
Technically this is also reading fields from the endpoint prior to grabbing a readlock, but it should grab the readlock first. Looking at the other similar `Handle()` functions, this function is a bit different in the ordering of endpoint readlock vs. annoCB(). Is there a reason that this code is not following the same pattern?

Reply:
You are right, I wasn't paying attention to the lock here. Fixed, also removed the logic that removes the rules when failed to grab lock, it should not be handled here anyways.